src/fhir_search/uri_parser.clj

```diff
@@ -0,0 +1,88 @@
+(ns fhir-search.uri-parser 
+  (:require [clojure.string :as str])
+  (:require [clojure.walk :refer [postwalk]])
+  (:import [java.net URI]))
+
+
+
+(defn url-encoding [url]
```

Comment:
I don't get the need for this fn? Are you demonstrating that java.net.URI already decodes the query-string when you call de `(.getQuery uri)` method? If so, that's great news, and lets remove this fn at all if we don't need it.